### PR TITLE
Implement Sprinter's Shoes

### PR DESCRIPTION
**Pull request by Wiggo32**
Who wants to run a little faster?

### CODE REVIEW

File: scripts/globals/items/sprinters_shoes.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    return 0;
+end;
+
+
+-----------------------------------------
+-- OnItemUse
+-----------------------------------------
+
+function onItemUse(target)
+    target:delStatusEffect(EFFECT_QUICKENING);
```
Comment (TeoTwawki) [inline scripts/globals/items/sprinters_shoes.lua]: no need to delStatusEffect, overwriting is baked into addStatusEffect and controlled by https://github.com/DarkstarProject/darkstar/blob/master/sql/status_effects.sql#L31

File: scripts/globals/items/sprinters_shoes.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    target:addStatusEffect(EFFECT_QUICKENING, 10, 0, 3600);
+    target:messageBasic(266,0, EFFECT_QUICKENING);
```
Comment (TeoTwawki) [inline scripts/globals/items/sprinters_shoes.lua]: adding the effect already should trigger this message. Even if it didn't,  we should
```lua
    GAINS_EFFECT           = 266, -- <target> gains the effect of <status>.
```
to the msg global instead off using "266" directly.

File: scripts/globals/items/sprinters_shoes.lua (hunk 1 above)
Comment (TeoTwawki) [reply scripts/globals/items/sprinters_shoes.lua]: https://github.com/DarkstarProject/darkstar/blob/master/src/map/status_effect.h#L34
```cpp
enum EFFECTOVERWRITE
{
    EFFECTOVERWRITE_EQUAL_HIGHER = 0, // only overwrite if equal or higher
    EFFECTOVERWRITE_HIGHER = 1, // only overwrite if higher
    EFFECTOVERWRITE_NEVER = 2, // never overwrite
    EFFECTOVERWRITE_ALWAYS = 3, // always overwrite no matter
    EFFECTOVERWRITE_IGNORE = 4 // ignore dupes
};
```

Comment (Wiggo32) [reply scripts/globals/items/sprinters_shoes.lua]: Ahhh, yes deleting that line still has the effect be overwritten. Nice!

File: scripts/globals/items/sprinters_shoes.lua (hunk 2 above)
Comment (Wiggo32) [reply scripts/globals/items/sprinters_shoes.lua]: However, I cannot get any of these items to trigger the message without the messageBasic line
https://github.com/DarkstarProject/darkstar/blob/2ca782d2c7cbdd3582827ea995c294d99595f176/scripts/globals/items/hermes_quencher.lua#L31
Maybe I'm doing something wrong? It triggers a message when the effect wears off though.

File: scripts/globals/items/sprinters_shoes.lua (hunk 1 above)
Comment (Wiggo32) [reply scripts/globals/items/sprinters_shoes.lua]: But, how do I deal with the conflict with things that have flee? 
http://ffxiclopedia.wikia.com/wiki/Powder_Boots
```
Notes: Does not overwrite or stack with effects of Sprinter's Shoes.
```

File: scripts/globals/items/sprinters_shoes.lua (hunk 2 above)
Comment (TeoTwawki) [reply scripts/globals/items/sprinters_shoes.lua]: I dunno seems to be missing globally, was sure it worked at one point..We even have a flag to suppress the msg in place..

Pull/rebase onto latest and then try this.

[sprinters_shoes.zip](https://github.com/DarkstarProject/darkstar/files/1420084/sprinters_shoes.zip)

File: scripts/globals/items/sprinters_shoes.lua (hunk 1 above)
Comment (TeoTwawki) [reply scripts/globals/items/sprinters_shoes.lua]: well those are 2 different effects. you just have one effect script check for the other (the effect, not the item) and behave accordingly. Or if the effect straight up won't apply that's even easier: we have an sql column for that.

File: scripts/globals/items/sprinters_shoes.lua (hunk 2 above)
Comment (Wiggo32) [reply scripts/globals/items/sprinters_shoes.lua]: Yes, that works now as I see you've added the GAINS_EFFECT msg.

**Comment by TeoTwawki** [comment]
Free weeks coming up on retail soon, merging as is and will cleanup anything off about it during free week

**Pull request merged**